src/components/CardContainer/CardContainer.tsx

```diff
@@ -1,16 +1,27 @@
-import * as React from "react";
+import React, { useEffect } from "react";
 import { Box, Grid } from "@mui/material";
 import { FilmCard } from "./FilmCard";
-import { cardData } from "../../mocks/cardData";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchPopularMovies } from "./api/services/fetchPopularMovies";
+import { Endpoints } from "./api/constants/endpoints";
 
 export const CardContainer = () => {
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchPopularMovies());
+  },[dispatch]);
+
+  const cardData = useSelector(state => state.cards.cards);
```

Comment:
1) add type for the selector
2) `state.cards.cards` is not readable, should be renamed

Reply:
2. done